tools/dagger_android_library.bzl

```diff
@@ -0,0 +1,54 @@
+# Copyright (C) 2020 The Dagger Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Macro for producing aars with Dagger's lint.jar in them.
+
+This works by basically invoking the native android_library rule under the hood, but then modifying
+its aar to embed the compiled lint.jar into it. To preserve build outputs integrity, the sole output
+of the genrule is just a lintpatch text file with an md5 hash of the final aar file.
+"""
+
+def dagger_android_library(name, **kwargs):
+    """An android_library with dagger's lint.jar.
+
+    Args:
+      name: The name of the target.
+    """
+    native_result = native.android_library(name = name, **kwargs)
```

Comment:
What if you use a suffix for the default target name, and then the genrule can just put the aar in the place you'd like it to be?

Reply:
That was my initial approach, but other android libraries that depend on this expect a particular naming structure for artifacts that would then no longer work for anything else generated by this rule

Reply:
What's an example of that?

Did you try using a genrule, and then wrapping that in an `aar_import` rule?

Reply:
One example is Android support will fail to fail due to trying to expose dagger Android. This seems to be a combination of depending on outputs it has plus the type of like libraries it can depend on.

I haven't tried that. I'm not sure I follow either 🤔. Before I go back down the rabbit hole, is there something inherently wrong with this implementation that merits using another one?

Reply:
I'm pretty sure modifying bazel outputs breaks the hermeticity model. In this case, you never really want someone to depend on the intermediate `android_library`, but in general Bazel rules create new outputs only.

Here's something like what I had in mind:

```python
native.android_library(
  name = name + "step_1",
  **kwargs
)

native.genrule(
  name = name + "inject_android_lint",
  ...
  outs = [name + "with_injected_android_lint.aar"],
)

native.aar_import(
  name = name + "aar_import",
  aar = name + "with_injected_android_lint.aar",
)

kwargs[deps] += [name + "aar_import"]
native.android_library(
  name = name, # this is the actual target people will "use"
  **kwargs
)
```